tests/cocotb/test_snitch_cc.py

```diff
@@ -0,0 +1,125 @@
+# ---------------------------------
+# Copyright 2023 KULeuven
+# Solderpad Hardware License, Version 0.51, see LICENSE for details.
+# SPDX-License-Identifier: SHL-0.51
+# Author: Ryan Antonio (ryan.antonio@esat.kuleuven.be)
+# ---------------------------------
+
+
+# -----------------------------------
+# Imports
+# -----------------------------------
+import subprocess
+import cocotb
+from cocotb.triggers import RisingEdge
+from cocotb.clock import Clock
+from cocotb_test.simulator import run
+import pytest
+
+
+# Reconfigurable parameters
+CLOCK_CYCLES = 20
+
+
+@cocotb.test()
+async def snitch_cc_dut(dut):
+    # Initialize clock
+    clock = Clock(dut.clk_i, 10, units="ns")
+    cocotb.start_soon(clock.start())
+
+    # Reset or intial values
+    dut.rst_ni.value = 0
+
+    # Wait 1 cycle to reset
+    await RisingEdge(dut.clk_i)
+
+    # Deassert reset
+    dut.rst_ni.value = 1
+
+    for i in range(CLOCK_CYCLES):
+        # Simple check if instructions are running normally
+        # instruction_value = hex(dut.i_snitch_cc.i_snitch.inst_data_i.value)
+        # instruction_address = int(dut.instruction_addr_offset.value)
+
+        # cocotb.log.info('---------- Instruction Info ----------')
+        # cocotb.log.info(f'Instruction Value: {instruction_value}')
+        # cocotb.log.info(f'Instruction Addr: {instruction_address}')
+
+        await RisingEdge(dut.clk_i)
+
+
+# Main test run
+@pytest.mark.parametrize("parameters", [{"AddrWidth": str(48), "DataWidth": str(64)}])
+def test_snitch_cc(parameters, simulator):
+    # Working paths
+    repo_path = "something"
+    tests_path = repo_path + "/tests/cocotb/"
+
+    filelist = subprocess.run(["bender", "script", "verilator"], stdout=subprocess.PIPE)
+
+    filelist = filelist.stdout.decode("utf-8").strip().split("\n")
+
+    include_folders = []
+    rtl_sources = []
+
+    for item in filelist:
+        if item == "":
+            pass
+        elif (
+            item[0] == "#"
+            or item[0] == ""
+            or item[0] == "\n"
+            or item[0:8] == "+define+"
+        ):
+            pass
+        elif item[0:8] == "+incdir+":
+            include_folders.append(item[8:].strip())
+        else:
+            rtl_sources.append(item.strip())
```

Comment:
A comment here would have been nice.
Also, it might be useful to create a common python module that you can import this functionality from, since it seems useful in more than just this testbench?

Reply:
Yes! Will add in a separate utils.py later. I had to experiment how to do this first but for information purposes I placed it "brute-force" way. Hmm maybe I can add a utils.py to this right now... let me think about it.

Reply:
Fixed this with commits [0867332](https://github.com/KULeuven-MICAS/snax-dev/pull/11/commits/0867332c941ebb90185cc3ad23d72eb78c0a9aca) and [33d4d13](https://github.com/KULeuven-MICAS/snax-dev/pull/11/commits/33d4d13a0db454a0ecc0a3c6174d55b8ecfcf5f8).